src/apphosting/secrets/dialogs.ts

```diff
@@ -0,0 +1,148 @@
+import * as clc from "colorette";
+const Table = require("cli-table");
+
+import * as apphosting from "../../gcp/apphosting";
+import * as prompt from "../../prompt";
+import * as iam from "../../gcp/iam";
+import * as utils from "../../utils";
+import { logger } from "../../logger";
+
+interface Metadata {
+  location: string;
+  id: string;
+  accounts: string[];
+}
+
+export function serviceAccountsForBackend(
```

Comment:
That function is not forwards compatible with explicit SAs and I'm not sure how to use the map of SAs when there will soon only be one.

My intention was to:
1. Finish the set command reasonably independently
2. Migrate all the existing source to appropraite locations
3. Reconcile the various commands

I still need to check with Abhi why the viewer permission is needed for Build. That may change things in the future.

Reply:
Hmmm... Given the fact that the build account needs an extra role I see what this was doing. I'll reconcile. This will be annoying and I need to go rehearse for I/O, so I'd like to get sign-off on what I have with a pinky promise to reconcile/fix this when I continue with the change tonight.

Reply:
sg, lets ship this for now and we can revisit later, esp much of this will change with the pending service account API updates